dropwizard-java8-jdbi/src/main/java/io/dropwizard/java8/jdbi/args/InstantArgument.java

```diff
@@ -0,0 +1,31 @@
+package io.dropwizard.java8.jdbi.args;
+
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.Argument;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.time.Instant;
+
+/**
+ * An {@link Argument} for {@link Instant} objects.
+ */
+public class InstantArgument implements Argument {
+
+    private Instant instant;
+
+    protected InstantArgument(Instant instant) {
+        this.instant = instant;
+    }
+
+    @Override
+    public void apply(int position, PreparedStatement statement, StatementContext ctx) throws SQLException {
+        if (instant != null) {
+            statement.setTimestamp(position, new Timestamp(instant.toEpochMilli()));
```

Comment:
Yes, I saw this way and just wanted to try it. Forgot to switch back.
Thanks
On Jul 16, 2015 7:41 PM, "Nick Babcock" <notifications@github.com> wrote:

> In
> dropwizard-java8-jdbi/src/main/java/io/dropwizard/java8/jdbi/args/InstantArgument.java
> <https://github.com/dropwizard/dropwizard-java8/pull/15#discussion_r34815512>
> :
>
> > +
> > +/**
> > + * An {@link Argument} for {@link Instant} objects.
> > + */
> > +public class InstantArgument implements Argument {
> > +
> > +    private Instant instant;
> > +
> > +    protected InstantArgument(Instant instant) {
> > +        this.instant = instant;
> > +    }
> > +
> > +    @Override
> > +    public void apply(int position, PreparedStatement statement, StatementContext ctx) throws SQLException {
> > +        if (instant != null) {
> > +            statement.setTimestamp(position, new Timestamp(instant.toEpochMilli()));
>
> Should it be Timestamp.from(instant)
> <https://docs.oracle.com/javase/8/docs/api/java/sql/Timestamp.html#from-java.time.Instant->
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropwizard/dropwizard-java8/pull/15/files#r34815512>.
>